### ISSUE TITLE
Can't use File objects when packaged as JAR

### ISSUE THREAD

**Issue by carterhudson** [ktorio/ktor#881]
I'm using my `resource` directory to store `html` files and image assets. I make the assets accessible as static content, like so: 

```kotlin
        ApplicationMain::class.java
            .getResource("html/assets")
            .toURI()
            .let { File(it) }
            .absolutePath
            .let {
                static("assets") {
                    files(it)
                }
            }
```
This works fine when I'm running from my IDE, but once I deploy it as a JAR file, the only way I can access resources are as `InputStream`s. However, much of the KTOR API requires `File` objects. I'm not sure how to proceed. I run into the same issue when I try to respond with `html` files like so: 
```kotlin
        route("/{page}") {
            get {
                ApplicationMain::class.java
                    .getResource("html/" + call.parameters["page"])
                    .toURI()
                    .let { uri ->
                        call.respondFile(File(uri))
                    }
            }
        }
```
Any ideas?

**Comment by cy6erGn0m** [comment]
Well, files inside of JAR are not represented by `java.io.File`. So you have two options: respond resource or unpack JAR at deployment

**Comment by carterhudson** [comment]
Is respond resource part of ktor api? I don't see a method like that anywhere. Can you clarify what you mean when you say respond resource?

**Comment by userquin** [comment]
see [Webjars](https://ktor.io/servers/features/webjars.html) feature

**Comment by userquin** [comment]
[Serving embedded resources](https://ktor.io/servers/features/static-content.html#serving-embedded-resources)

**Comment by carterhudson** [comment]
> [Serving embedded resources](https://ktor.io/servers/features/static-content.html#serving-embedded-resources)

Thanks for this! I think that might solve one problem. 

I'm still having issues sending back html responses. I found a method `ApplicationCall.resolveResource(...)` that I think may do what I need (retrieve html file from resources), but it's returned as an `OutgoingContent`. I'm not sure how to turn this into a useful response.

**Comment by userquin** [comment]
you dont need to do anything, just configure:

```java
resources("/pages")
```

and when you request `/somepage.html` if `somepage.html` exists in classpath `/pages` directory  you are done. You dont have to map each resource, just add directories like this:

- /pages
  - somepage.html
  - somepage.html
- /css
  - style.css

and then add `resources("pages")` and `resources("css")`or just use `staticRootFolder`or a combination

**Comment by userquin** [comment]
you can see [Static Sample](https://ktor.io/samples/feature/static.html) and how is configured to serve css for examples.

**Comment by carterhudson** [comment]
Ended up having to un-nest my resource sub folders. I had `resources/html` with all my html files, and `resources/html/assets` with all of my image assets. I couldn't get it to serve any of the assets when I did `resources("html/assets")` or `resources("/html/assets")`. I reorganized them to `resources/html` and `resources/assets` and now it all seems to be working. I really appreciate your help!
